dagshub/data_engine/client/loaders/base.py

```diff
@@ -81,7 +81,22 @@ def _get_file_columns(self):
             [self.entries[0].metadata[col] for col in self.metadata_columns],
         ):
             try:
-                self.repo.list_path((self.datasource_root / str(value)).as_posix())
+                if self.source == "repo":
+                    self.repo.list_path((self.datasource_root / str(value)).as_posix())
+                else:
+                    self.repo.list_storage_path(
+                        (
+                            Path(
+                                "/".join(
+                                    list(self.datasource.source.path_parts().values())[
+                                        :2
+                                    ]
+                                )
+                            )
+                            / self.datasource_root
+                            / str(value)
+                        ).as_posix()
```

Comment:
I think I already did this same thing once.
Check out `self.datasource.source.root_content_path`. This will give you `https://dagshub.com/api/v1/repos/jinen/test-repo/storage/content/s3/bucket/`, and then you'll need to add the path of the datapoint itself to it.
The function also handles repos and buckets, so no need for the if/else
https://github.com/DagsHub/client/blob/11f90c340779b7d4d4fb584b8c86662cbb10f556/dagshub/data_engine/model/datasource_state.py#L124

Reply:
Wait, even better, there's a `self.datasource.source.content_path`, that also appends the path!
https://github.com/DagsHub/client/blob/11f90c340779b7d4d4fb584b8c86662cbb10f556/dagshub/data_engine/model/datasource_state.py#L89

Reply:
Awesome, thank you!

Reply:
I just realized, this won't actually work, since the idea behind it is not getting the value, but checking if it exists in the remote. Those functions get me the target url, but obtaining the final path (just from 's3/bucket/path/to/file') still requires string manipulation similar to what's already on.

I cannot directly pass the URL to `list_storage_path` or `list_path` in RepoAPI, and I do not want to send queries within the dataloader class because of DRY and I'll have to manage authentication.

Reply:
The function I wrote just generates the path though, it doesn't do a request. Can you demonstrate an example where you know it won't work for some reason?